library/vyos_facts.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The module file for vyos_facts
+"""
+
+from __future__ import absolute_import, division, print_function
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.connection import Connection
+from ansible.module_utils. \
+     vyos.facts.facts import Facts
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': [u'preview'],
+                    'supported_by': 'network',}
+
+
+DOCUMENTATION = """
+---
+module: vyos_facts
+version_added: 2.9
+short_description: Get facts about vyos devices.
+description:
+  - Collects facts from network devices running the vyos operating
+    system. This module places the facts gathered in the fact tree keyed by the
+    respective resource name.  The facts module will always collect a
+    base set of facts from the device and can enable or disable
+    collection of additional facts.
+author: Nilashish Chakraborty (@nilashishc)
+options:
+  gather_subset:
+    description:
+      - When supplied, this argument will restrict the facts collected
+        to a given subset. Possible values for this argument include
+        all, min, hardware, config, legacy, and interfaces. Can specify a
+        list of values to include a larger subset. Values can also be used
+        with an initial C(M(!)) to specify that a specific subset should
+        not be collected.
+    required: false
+    default: 'all'
+    version_added: "2.2"
+  gather_network_resources:
+    description:
+      - When supplied, this argument will restrict the facts collected
+        to a given subset. Possible values for this argument include
+        all and the resources like interfaces, vlans etc.
```

Comment:
Same comment on potentially making this choices:

Reply:
Same as above